src/miniscript/context.rs

```diff
@@ -94,6 +111,12 @@ impl ScriptContext for Legacy {
     ) -> Result<(), ScriptContextError> {
         Ok(())
     }
+
+    fn check_ms_validity<Pk: MiniscriptKey, Ctx: ScriptContext>(
```

Comment:
I think `check_ms_validity` and `check_frag_validity` can be merged. The only thing the `check_frag_validity` is doing is checking uncompressed public keys in Segwit Context.

Reply:
There also needs to be check in the FromStr for Miniscript. But I think if we combine both of these, the call sites are automatically covered.

Reply:
> I think check_ms_validity and check_frag_validity can be merged. The only thing the check_frag_validity is doing is checking uncompressed public keys in Segwit Context.

I thought too and actually that's how i implemented it at first. Turns out it's quite messy (and costy) as we need to match the terminal and duplicate the logic, better to call it on the actual top miniscript.